File: BPM.Web.API/Models/Entities/Warehouse.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace BPM.Web.API.Models.Entities
{
    [Table("warehouses")]
    public class Warehouse
    {
        [Key]
        [Column("id")]
        public Guid Id { get; set; }

        [Required]
        [MaxLength(50)]
        [Column("warehousecode")]
        public string WarehouseCode { get; set; } = string.Empty;

        [Required]
        [MaxLength(200)]
        [Column("warehousename")]
        public string WarehouseName { get; set; } = string.Empty;

        [Column("distributorid")]
        public Guid? DistributorId { get; set; }

        [MaxLength(255)]
        [Column("addressline1")]
        public string? AddressLine1 { get; set; }

        [MaxLength(255)]
        [Column("addressline2")]
        public string? AddressLine2 { get; set; }

        [MaxLength(100)]
        [Column("city")]
        public string? City { get; set; }

        [MaxLength(100)]
        [Column("state")]
        public string? State { get; set; }

        [MaxLength(100)]
        [Column("country")]
        public string? Country { get; set; }

        [MaxLength(20)]
        [Column("postalcode")]
        public string? PostalCode { get; set; }

        [Column("isactive")]
        public bool IsActive { get; set; }

        [Column("createdby")]
        public Guid? CreatedBy { get; set; }

        [Column("createdon")]
        public DateTime CreatedOn { get; set; }

        [Column("modifiedby")]
        public Guid? ModifiedBy { get; set; }

        [Column("modifiedon")]
        public DateTime? ModifiedOn { get; set; }
    }
}

File: BPM.Web.Dealer.UI/src/app/models/warehouse.ts
export interface Warehouse {
  id?: string;
  warehouseCode?: string;
  warehouseName?: string;
  distributorId?: string;
  addressLine1?: string;
  addressLine2?: string;
  city?: string;
  state?: string;
  country?: string;
  postalCode?: string;
  isActive?: boolean;
  createdBy?: string;
  createdOn?: string;
  modifiedBy?: string;
  modifiedOn?: string;
}

export interface WarehouseCreateDto {
  warehouseCode: string;
  warehouseName: string;
  distributorId?: string;
  addressLine1?: string;
  addressLine2?: string;
  city?: string;
  state?: string;
  country?: string;
  postalCode?: string;
  createdBy?: string;
}

export interface WarehouseUpdateDto {
  id: string;
  warehouseName: string;
  distributorId?: string;
  addressLine1?: string;
  addressLine2?: string;
  city?: string;
  state?: string;
  country?: string;
  postalCode?: string;
  isActive: boolean;
  modifiedBy?: string;
}
